tests/phpunit/tests/blocks/renderHooks.php

```diff
@@ -0,0 +1,61 @@
+<?php
+/**
+ * Tests for the hooks of the block rendering functions.
+ *
+ * @package WordPress
+ * @subpackage Blocks
+ * @since 6.4.0
+ *
+ * @group blocks
+ */
+class Tests_Blocks_Render_Hooks extends WP_UnitTestCase {
+	/**
+	 * @ticket 59131
+	 */
+	public function test_do_blocks_pre_render_filter() {
+		$remove_blocks_containing_wordpress = static function( $blocks ) {
+			foreach ( $blocks as $index => $block ) {
+				if ( strpos( $block['innerHTML'], 'WordPress' ) !== false ) {
+					unset( $blocks[ $index ] );
+				}
+			}
+			return $blocks;
+		};
+
+		$block_content = '
+			<!-- wp:core/paragraph --><p>Hello</p><!-- /wp:core/paragraph -->
+			<!-- wp:core/paragraph --><p>WordPress</p><!-- /wp:core/paragraph -->
+		';
+
+		add_filter( 'do_blocks_pre_render', $remove_blocks_containing_wordpress );
+
+		$result = do_blocks( $block_content );
+
+		remove_filter( 'do_blocks_pre_render', $remove_blocks_containing_wordpress );
+
+		$this->assertSame( '<p>Hello</p>', trim( $result ) );
```

Comment:
if the paragraph renders differently then this test might fail when it shouldn't. if we wanted to make the tests more resilient, we could separate the block content and run one manually without the excluded blocks, that way it would be solely testing the filter and not also the block rendering code.

```php
$hello_block = '<!-- wp:core/paragraph --><p>Hello</p><!-- /wp:core/paragraph -->';
$press_block = '<!-- wp:core/paragraph --><p>WordPress</p><!-- /wp:core/paragraph -->';

add_filter( 'do_blocks_pre_render', $remove_blocks_containing_wordpress );
$filtered_output = do_blocks( $hello_block . $press_block );
remove_filter( 'do_blocks_pre_render', $remove_blocks_containing_wordpress );

$wanted_output = do_blocks( $hello_block );
$this->assertSame( $wanted_output, $filtered_output, "Did not exclude the intended blocks before rendering." );
```